contribs/gmf/src/directives/mobilemeasurelength.js

```diff
@@ -0,0 +1,216 @@
+ * @ngdoc directive
+ * @ngname gmfMobileMeasureLength
+ */
+gmf.mobileMeasureLengthDirective =
+    function(gmfMobileMeasureLengthTemplateUrl) {
+  return {
+    restrict: 'A',
+    scope: {
+      'map': '=gmfMobileMeasureLengthMap',
+      'mobileDrawLine': '=gmfMobileMeasureLengthDrawLine'
+    },
+    controller: 'GmfMobileMeasureLengthController',
+    controllerAs: 'measureLengthCtrl',
+    bindToController: true,
+    templateUrl: gmfMobileMeasureLengthTemplateUrl
+  };
+
+};
+
+
+gmfModule.directive('gmfMobileMeasureLength',
+                    gmf.mobileMeasureLengthDirective);
+
+
+
+/**
+ * @param {angularGettext.Catalog} gettextCatalog Gettext catalog.
+ * @param {!angular.Scope} $scope Angular scope.
+ * @param {angular.$compile} $compile Angular compile service.
+ * @param {angular.$sce} $sce Angular sce service.
+ * @constructor
+ * @ngInject
+ * @ngdoc controller
+ * @ngname GmfMobileMeasureLengthController
+ */
+gmf.MobileMeasureLengthController = function(gettextCatalog, $scope, $compile,
+    $sce) {
+
+  /**
+   * @type {ol.Map}
+   * @export
+   */
+  this.map;
+
+  /**
+   * @type {angularGettext.Catalog}
+   * @private
+   */
+  this.gettextCatalog_ = gettextCatalog;
+
+  /**
+   * @type {ngeo.interaction.MobileDraw}
+   * @export
+   */
+  this.mobileDrawLine;
+
+  var mobileDrawLine = this.mobileDrawLine;
+
+  Object.defineProperty(this, 'hasPoints', {
+    get: function() {
+      return !goog.isNull(this.mobileDrawLine.getFeature());
+    }
+  });
+
+  /**
+   * @type {boolean}
+   * @export
+   */
+  this.dirty = false;
+
+  /**
+   * @type {boolean}
+   * @export
+   */
+  this.drawing = false;
+
+  /**
+   * @type {boolean}
+   * @export
+   */
+  this.valid = false;
+
+  /**
+   * @type {boolean}
+   * @export
+   */
+  this.active = false;
+
+  ol.events.listen(
+      mobileDrawLine,
+      ol.Object.getChangeEventType(
+          ngeo.interaction.MobileDrawProperty.DIRTY),
+      function() {
+        this.dirty = mobileDrawLine.getDirty();
+
+        // this is where the angular scope is forced to be applied. We
+        // only need to do this when dirty, as going to "no being dirty"
+        // is made by a click on a button where Angular is within scope
+        if (this.dirty) {
+          $scope.$apply();
+        }
+      },
+      this
+  );
+
+  ol.events.listen(
+      mobileDrawLine,
+      ol.Object.getChangeEventType(
+          ngeo.interaction.MobileDrawProperty.DRAWING),
+      function() {
+        this.drawing = mobileDrawLine.getDrawing();
+      },
+      this
+  );
+
+  ol.events.listen(
+      mobileDrawLine,
+      ol.Object.getChangeEventType(
+          ngeo.interaction.MobileDrawProperty.VALID),
+      function() {
+        this.valid = mobileDrawLine.getValid();
+      },
+      this
+  );
+
+  ol.events.listen(
+      mobileDrawLine,
+      ol.Object.getChangeEventType(ol.interaction.InteractionProperty.ACTIVE),
+      function() {
+        this.active = mobileDrawLine.getActive();
```

Comment:
We discussed about this already, remember ?  I had a decorator, but I had issues with the fact that the mobileDrawLine is an ol3 control and it doesn't natively contain an `active` property.  Well, it shouldn't because it already has an inner property set with the `set` method.

We decided together that we were to remove the decorator and put the logic in the directive instead.  That point is still valid, in my opinion.  Please, validate.